packages/dev/core/src/scene.ts

```diff
@@ -5577,8 +5728,8 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
      * @returns the texture or null if none found.
      * @deprecated Please use getTextureByUniqueId instead
      */
-    getTextureByUniqueID(uniqueId: number): Nullable<BaseTexture> {
-        return this.getTextureByUniqueId(uniqueId);
+    getTextureByUniqueID<T extends BaseTexture>(uniqueId: number): Nullable<T> {

```

Comment:
This one needs overloading

Reply:
this method is deprecated, I'm not going to declare generic type for it, so the method declaration will be kept the same as before, no generic type.